File: 617 - Merge Two Binary Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def mergeTrees(self, root1: TreeNode, root2: TreeNode) -> TreeNode:
        if root1 is None and root2 is None:
            return None
        root = TreeNode()
        if root1 is not None:
            root.val += root1.val
        if root2 is not None:
            root.val += root2.val
        root.left = self.mergeTrees(root1.left if root1 is not None else None, root2.left if root2 is not None else None)
        root.right = self.mergeTrees(root1.right if root1 is not None else None, root2.right if root2 is not None else None)
        return root

File: 617 - Merge Two Binary Trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {TreeNode}
 */
var mergeTrees = function (root1, root2) {
    if (root1 == null && root2 == null) {
        return null;
    }
    let root = new TreeNode();
    if (root1 != null) {
        root.val += root1.val;
    }
    if (root2 != null) {
        root.val += root2.val;
    }
    root.left = mergeTrees(root1 == null ? null : root1.left, root2 == null ? null : root2.left);
    root.right = mergeTrees(root1 == null ? null : root1.right, root2 == null ? null : root2.right);
    return root;

};

console.dir();